### ISSUE TITLE
how to upload bmp?

### ISSUE THREAD

**Issue by polipen** [AlexALX/wiremod_e2_os#18]
i tried imgur, my ftp server, some online image viewers but not worked.

**Comment by AlexALX** [comment]
you need own web (http) server, not ftp one. Most image services like imgur converting bmp to jpg always, so you can't upload raw (true) bmp image here.

**Comment by polipen** [comment]
its now giving me "empty response" error and yes i made my localhost server and its an bmp image

**Comment by AlexALX** [comment]
Did you tried open url in browser? And make sure its not httpS but just http (since i had issues with https in wiremod)

**Comment by polipen** [comment]
yes its http and yes i can open the image in my browser. btw i'm using iis feature of windows.

**Comment by AlexALX** [comment]
Then i'm not sure, try 127.0.0.1 instead of localhost, also it might be related with this warning: https://wiki.facepunch.com/gmod/http.Post

> To enable HTTP-requests to destinations on private networks use [Command Line Parameters](https://wiki.facepunch.com/gmod/Command_Line_Parameters) -allowlocalhttp (serverside only).

EDIT: just checked my gmod params - and yes i did added that at some point and forgot (so just add param in steam and will work for singleplayer).

**Comment by polipen** [comment]
parameter is worked thanks! and last thing, is there a way to make bigger dhdd? i mean i can't get songs its so small

when i try to edit hdd e2 chip it can't validate the code (failed to include "alx_pc/shared/_main_") when i delete it, it says "no such function: getentity(number).

**Comment by AlexALX** [comment]
Yes you can make bigger dhdd by adding extra data banks and then edit e2 chip with adding # of them + connect with wirelink. As for error - you need download e2 folder manually (from this repository) and extract so path will be valid, then use main script (alx_pc/HDD.txt) when upload to e2 chip.

And note - 1 MB is physical limit for each HDD Bank (262144 sectors with 32bits).

**Comment by polipen** [comment]
ok last thing (really) a glitch happening to my pictures
![20220928163747_1](https://user-images.githubusercontent.com/109471182/192793859-9eb6bacb-e00e-4a4d-ae38-b14cb4937595.jpg)

it should be complete white (look closer if u didn't see it)

**Comment by AlexALX** [comment]
Are you sure its not rendering issue (like scaling from different distance), or conversion issue (like from jpg to bmp)?

**Comment by AlexALX** [comment]
Also make sure you didn't enabled "Compression" in any kind when save bmp. it should be 32/24/16/8/4/1 bits without compression. Also doesn't support RTL or something like this called, not remember already.
